core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt

```diff
@@ -0,0 +1,31 @@
+package fr.sncf.osrd.sim_infra.api
+
+import fr.sncf.osrd.railjson.schema.geom.LineString
+import fr.sncf.osrd.utils.indexing.DirStaticIdxList
+import fr.sncf.osrd.utils.indexing.StaticIdx
+import fr.sncf.osrd.utils.units.Distance
+import fr.sncf.osrd.utils.DistanceRangeMap
+
+data class Path(
+    val chunks: DirStaticIdxList<TrackChunk>,
+    val beginOffset: Distance,
+    val endOffset: Distance,
+)
+
+data class WithOffset<T>(
+    val value: T,
+    val offset: Distance,
+)
+typealias IdxWithOffset<T> = WithOffset<StaticIdx<T>>
+
+interface PathProperties {
+    fun getSlopesOnPath(path: Path): DistanceRangeMap<Double>
+    fun getOperationalPointsOnPath(path: Path): List<IdxWithOffset<OperationalPoint>>
```

Comment:
I'm a little confused, why is it needed again?

Reply:
It will be used in `PathfindingResultConverter`